File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int profit = 0;
        int n = prices.length;
        
        for(int i=0; i<n; i++){
            if(prices[i] < min){
                min = prices[i];
            }
            if(prices[i] - min > profit){
                profit = prices[i] - min;
            }
        }
        
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
function maxProfit(prices) {
    let min = prices[0];
    let max = prices[0];
    let total = 0;
    for(let z of prices){
        if(z < min){
            min = z;
            max = z;
            continue;
        }

        if(z > max){
            max = z;
            total = Math.max(total, max - min)
        }
    }

    return total;
};
